qiskit/transpiler/passes/calibration/rx_builder.py

```diff
@@ -30,8 +30,25 @@ class RXCalibrationBuilder(CalibrationBuilder):
     """Add single-pulse RX calibrations that are bootstrapped from the SX calibration.
 
     .. note::
+
         Requirement: NormalizeRXAngles pass (one of the optimization passes).
 
+    It is recommended to place this pass in the post-optimization stage of a passmanager.
+    A simple demo:
+
+    .. code-block:: python
+
+       backend = FakeBelemV2()
+       pm = PassManager(RXCalibrationBuilder(backend.target))
```

Comment:
Yes, it would be good to have a complete example for running a circuit on a backend. One typically wants to call `transpile()` on the circuit first but the rx gates will confuse `transpile()` unless you run `RXCalibrationBuilder` first or integrate it into the default pass manager.

Reply:
This example shows that this pass decreases the gate count of a QuantumVolume circuit.
- Gate count with this pass: `[('rz', 93), ('rx', 48), ('cx', 28)]`
- Gate count without this pass: `[('rz', 97), ('sx', 68), ('cx', 31)]`

``` python
# This is the code snippet for the demo
# Assumed that NormalizeRXAngles pass is already in the normal import path
from qiskit.providers.fake_provider import FakeBelemV2
from qiskit.transpiler import PassManager, PassManagerConfig
from qiskit.transpiler.preset_passmanagers import level_1_pass_manager
from qiskit.circuit import Parameter
from qiskit.circuit.library import QuantumVolume
from qiskit.circuit.library.standard_gates import RXGate

from calibration.rx_builder import RXCalibrationBuilder

qv = QuantumVolume(4, 4, seed=1004)

# Transpiling with single pulse RX gates enabled
backend_with_single_pulse_rx = FakeBelemV2()
rx_inst_props = {}
for i in range(backend_with_single_pulse_rx.num_qubits):
    rx_inst_props[(i,)] = None
backend_with_single_pulse_rx.target.add_instruction(RXGate(Parameter("theta")), rx_inst_props)
config_with_rx = PassManagerConfig.from_backend(backend=backend_with_single_pulse_rx)
pm_with_rx = level_1_pass_manager(pass_manager_config=config_with_rx)
rx_builder = RXCalibrationBuilder(target=backend_with_single_pulse_rx.target)
pm_with_rx.post_optimization = PassManager([rx_builder])
transpiled_circ_with_single_pulse_rx = pm_with_rx.run(qv)
transpiled_circ_with_single_pulse_rx.count_ops()

# Conventional transpilation: each RX gate is decomposed into a sequence with two SX gates
original_backend = FakeBelemV2()
original_config = PassManagerConfig.from_backend(backend=original_backend)
original_pm = level_1_pass_manager(pass_manager_config=original_config)
original_transpiled_circ = original_pm.run(qv)
original_transpiled_circ.count_ops()
```

But please notice that the version below works while the `NormalizeRXPass` is not yet distributed as part of Qiskit.
``` python
from qiskit.providers.fake_provider import FakeBelemV2
from qiskit.transpiler import PassManager, PassManagerConfig
from qiskit.transpiler.preset_passmanagers import level_1_pass_manager
from qiskit.circuit import Parameter
from qiskit.circuit.library import QuantumVolume
from qiskit.circuit.library.standard_gates import RXGate

from calibration.rx_builder import RXCalibrationBuilder  # NOTE: modify this according to the location of this snippet

qv = QuantumVolume(4, 4, seed=1004)
```

``` python
# importing the modified qiskit library
# because the script for RXCalibrationBuilder does
# "from qiskit.transpiler.passes.optimization import NormalizeRXAngles" (line 69)
# but NormalizeRXAngles is available only in my local development branch.
# https://stackoverflow.com/questions/67631/how-can-i-import-a-module-dynamically-given-the-full-path

import importlib
import sys
MODULE_PATH = "/Users/jaeunkim/Code/qiskit-terra/qiskit/transpiler/passes/optimization/__init__.py"
MODULE_NAME = "qiskit.transpiler.passes.optimization"
spec = importlib.util.spec_from_file_location(MODULE_NAME, MODULE_PATH)
module = importlib.util.module_from_spec(spec)
sys.modules[spec.name] = module 
spec.loader.exec_module(module)
```

``` python
# Transpiling with single pulse RX gates enabled
backend_with_single_pulse_rx = FakeBelemV2()
rx_inst_props = {}
for i in range(backend_with_single_pulse_rx.num_qubits):
    rx_inst_props[(i,)] = None
backend_with_single_pulse_rx.target.add_instruction(RXGate(Parameter("theta")), rx_inst_props)
config_with_rx = PassManagerConfig.from_backend(backend=backend_with_single_pulse_rx)
pm_with_rx = level_1_pass_manager(pass_manager_config=config_with_rx)
rx_builder = RXCalibrationBuilder(target=backend_with_single_pulse_rx.target)
pm_with_rx.post_optimization = PassManager([rx_builder])
transpiled_circ_with_single_pulse_rx = pm_with_rx.run(qv)
transpiled_circ_with_single_pulse_rx.count_ops()
```

``` python
# Conventional transpilation: each RX gate is decomposed into a sequence with two SX gates
original_backend = FakeBelemV2()
original_config = PassManagerConfig.from_backend(backend=original_backend)
original_pm = level_1_pass_manager(pass_manager_config=original_config)
original_transpiled_circ = original_pm.run(qv)
original_transpiled_circ.count_ops()
```